miner/worker.go

```diff
@@ -378,13 +387,20 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 		case <-w.startCh:
 			clearPending(w.chain.CurrentBlock().NumberU64())
 			timestamp = time.Now().Unix()
-			commit(false, commitInterruptNewHead)
+			commit(false, commitInterruptNewHead, true)
 
-		case head := <-w.chainHeadCh:
-			headNumber := head.Block.NumberU64()
+		case <-w.chainHeadCh:
+			if h, ok := w.engine.(consensus.Handler); ok {
+				h.NewWork()
```

Comment:
I think `NewWork` should be renamed to `NewChainHead`.

Reply:
There is an existing NewChainHead in `consensus.go`: https://github.com/celo-org/celo-blockchain/blob/83cb6f6b8d5777059f68824fb5d3ffc59c579721/consensus/consensus.go#L141

Reply:
Good catch.  Looks like this function https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/backend/handler.go#L276 will need to be modified so that the final committed event is sent to the core consensus engine and the `NewWork` function can be removed altogether.